### DISCUSSION TITLE
Advanced Relationship tables

### DISCUSSION THREAD

**Discussion by lsmith212a** [pimcore/pimcore#13814] [Q&A / Support] [answered]
I am try to write SQL to get at the object_meta_data_(id) tables, but I cannot figure out what ID I need to use.

For example I have an object COV
One of the objects in the COV table is ID 65
In the object_relations_COV table, id 18 defines a relationship between src_id 65 and dest_id 31
I want to find the extra data in that relationship. I see that its object_meta_data_(id) but I can't seem to find any table using any of these IDs. What ID does it want?

Appreciate any help you can give.

**Comment by mattamon** [comment]
Hey @lsmith212a,

If I do understand it correctly you want extra information for the COV object with the ID 65 right?
The important ID the dest_id here. 

The dest_id is bascially the ID of the related object. So you can then use the ID 31 to get more information for the COV Object with the ID 65.

Please correct me if am wrong and this is not what you wanted :)

**Comment by lsmith212a** [comment, marked as answer]
Close. I'm actually looking for the data that would be in the object_meta_data_(id) table, but I don't know which id it's looking for. This is where I understand the data entered for the extra columns defined on an advanced many-to-many relationship field.

In object 65 I have a relationship with the 31 object and there are a few extra data fields defined in that relationship. That's the data I'm trying to find.

![image](https://user-images.githubusercontent.com/57369009/208242228-6d26bd69-94b8-41e6-aba6-c39cb4b76be8.png)


Thank you for your response

**Reply by mattamon** [reply]
Hey @lsmith212a,

ah now I see what you're looking for! 

Well first of all, when you are creating a new class you can give it a class name and a unique identifier. 
![image](https://user-images.githubusercontent.com/38959016/208367446-47df54ff-6b4d-4ac3-bdc7-62b06375a9b4.png)

The unique identifier can be a number or a text or a combination. 
In the demo we often use names as unique identifiers, like e.g. the customer class would be CU.
![image](https://user-images.githubusercontent.com/38959016/208367648-b6a97cac-73d2-47ae-ac91-f5183f9ac692.png)

The ```object_metadata``` table would then be named ```object_metadata_CU```

To find the right ID you have to go to your class definitions or if you want to do it in the Code you can use the classId method of the object to get the right ID.

```php
$classId = Customer::classId()
```

From there on it should be pretty straight forward.
The o_id would be 65 in your case and the dest_id would be 31
![image](https://user-images.githubusercontent.com/38959016/208368455-0a0d0285-9f6d-41bd-af27-b8b3b0eeaceb.png)

**Comment by lsmith212a** [comment]
I got it to work! Thank you! I was trying to use object_meta_data_(id) per the documentation [on this page](https://pimcore.com/docs/pimcore/10.1/Development_Documentation/Development_Tools_and_Details/Database_Model.html), but I noticed in your response that you used object_metadata_(id). When I tried that I was able to write a custom report against that table using the ID for the class.
